### DISCUSSION TITLE
CreateDoubleHistogram with no buckets

### DISCUSSION THREAD

**Discussion by ashley-b** [open-telemetry/opentelemetry-cpp#3222] [Q&A] [answered]
Hi,

Is it possible to have CreateDoubleHistogram with no bucket and just use the min and max collected by this Instrument.

This would be useful for capturing quickly changing metric values and seeing any extreme values above or below the mean or last value. Currently am using a gauge, but this only uses the last value as I understand it.

**Comment by malkia** [comment, marked as answer]
You can add a view, for example: 

```c++
  {
      using namespace opentelemetry::sdk::metrics;
      auto aggr_conf = std::make_unique<HistogramAggregationConfig>();  
      aggr_conf->boundaries_ = std::vector<double>{0.0f};
      aggr_conf->record_min_max_ = false;
      provider->AddView(
        InstrumentSelectorFactory::Create(InstrumentType::kHistogram, "*", ""), 
        MeterSelectorFactory::Create("", "", ""),
        ViewFactory::Create( "", "", "", AggregationType::kHistogram, std::move( aggr_conf ) )
      );
  }
```

This creates only two buckets - 0.0, and inf+

**Reply by ashley-b** [reply]
Thanks @malkia, that works well.
One thing I needed to change was to have min and max recorded with `aggr_conf->record_min_max_ = true;`

**Reply by malkia** [reply]
Oh, thanks now to you - as I may have to add this on my end!
